fluss-flink/fluss-flink-common/src/main/java/org/apache/fluss/flink/source/FlinkTableSource.java

```diff
@@ -321,6 +342,36 @@ public boolean isBounded() {
         }
     }
 
+    private boolean pushTimeStampFilterToLakeSource(
+            LakeSource<?> lakeSource, RowType flussRowType) {
+        // will push timestamp to lake
+        // we will have three additional system columns, __bucket, __offset, __timestamp
+        // in lake, get the  __timestamp index in lake table
+        final int timestampFieldIndex = flussRowType.getFieldCount() + 2;
+        Predicate timestampFilter =
+                new LeafPredicate(
+                        GreaterOrEqual.INSTANCE,
+                        DataTypes.TIMESTAMP_LTZ(),
+                        timestampFieldIndex,
+                        TIMESTAMP_COLUMN_NAME,
+                        Collections.singletonList(
+                                TimestampLtz.fromEpochMillis(startupOptions.startupTimestampMs)));
+        List<Predicate> acceptedPredicates =
+                lakeSource
+                        .withFilters(Collections.singletonList(timestampFilter))
+                        .acceptedPredicates();
+        if (acceptedPredicates.isEmpty()) {
+            LOG.warn(
+                    "The lake source doesn't accept the filter {}, won't read data from lake.",
+                    timestampFilter);
+            return false;
+        }
+        checkState(
+                acceptedPredicates.size() == 1
+                        && acceptedPredicates.get(0).equals(timestampFilter));
```

Comment:
Why is an exception being thrown here? If simply check if the `timestampFilter` exists, and if it doesn't, return false and log an error, is ok?

Reply:
It means, we pass a filter to lake, and lake accept the filter, but the filter is not the passed filter. It should never happen, if it happens,   it must be that there is a bug in lake source implementation. 
So, I think we should throw exception directly to fast fail.